torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1406,59 +1443,6 @@ std::string ProcessGroupNCCL::getNCCLWatchdogDebugInfo() {
   return retrieveDesyncReport(store_, "NCCL", rank_, size_);
 }
 
-#if defined(__linux__)
-struct DumpPipe {
```

Comment:
Move the declaration to the header file so that heartbeatMonitor() function can access it, defining in cpp file will make the access depending on the order of function implementation